forward/forwarder_test.go

```diff
@@ -188,6 +293,85 @@ func (s *ForwarderTestSuite) TestRequestNoReroutes() {
 	s.Error(err)
 }
 
+func (s *ForwarderTestSuite) TestRegisterListener() {
+	listener := &EventListener{}
+	listener.On("HandleEvent").Return()
+
+	s.forwarder.RegisterListener(listener)
+	s.Assertions.Equal(1, len(s.forwarder.listeners), "Expected 1 listener to be registered")
+
+	// remove all listeners
+	s.forwarder.listeners = nil
```

Comment:
I'm not really sure, but I was getting unexpected events to the stub after the test was ran. If there are calls that don't have a mocked response mock/mockery panics and the test crashes.

I have validated that a new suite will be created for every test run so why the unexpected events are emitted is unclear to me.

I also tried removing all listeners in the `TearDown` routine, but this still resulted in panics.